hnn_core/network.py

```diff
@@ -570,6 +593,9 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
         target_populations, weights_ampa, weights_nmda = \
             _get_target_populations(weights_ampa, weights_nmda)
 
+        if len(target_populations) == 0:
+            warn('No target populations have been specified for this drive.')
```

Comment:
error, no? We don't want ghost drives that only affect the random state ...

Reply:
This will cause loading drives from params in most of our param files to throw an error due to the fact that our param files (e.g., `default.json`) each contain silent gaussian and poisson drives who's `target_populations` get removed in [`params._extract_drive_specs_from_hnn_params()`](https://github.com/jonescompneurolab/hnn-core/blob/40b07726a747c336debfe3a2683ee1607391a75d/hnn_core/params.py#L244). We could remove that `if` statement?

Reply:
If the connections to all the target populations has 0 weight in the param file, is the desired behavior to 1) keep the drive with connection weight of 0, or 2) remove the drive? If they want to remove the drive, should they remove the keys for that drive? Either way, I think the logic should be inside `extract_drive_specs_from_hnn_params` since it's related to how the param file should be handled.

Reply:
That makes sense. I'll modify this to throw an error instead of warning and adjust the param file parsing accordingly.

Reply:
I agree. A lot of code can be simplified now that we don't have to respect poor implementations of the past.